.github/actions/deploy-backend/action.yml

```diff
@@ -0,0 +1,103 @@
+        echo "ACR_NAME=rkh5012containerregistry.azurecr.io" >> $GITHUB_ENV
+        echo "IMAGE_REPO=rkh5012" >> $GITHUB_ENV
+        echo "TAG=${{ github.sha }}" >> $GITHUB_ENV
+        echo "FUNCTION_APP=rkh5012-functionapp" >> $GITHUB_ENV
+
+    - name: Set environment variables - test
+      if: github.ref == 'refs/heads/ronheft/test-investigation'
+      shell: bash
+      run: |
+        echo "RESOURCE_GROUP=prime-data-hub-test" >> $GITHUB_ENV
+        echo "TERRAFORM_ENV=test" >> $GITHUB_ENV
+        echo "ACR_NAME=pdhtestcontainerregistry.azurecr.io" >> $GITHUB_ENV
+        echo "IMAGE_REPO=pdhtest" >> $GITHUB_ENV
+        echo "TAG=${{ github.sha }}" >> $GITHUB_ENV
+        echo "FUNCTION_APP=pdhtest-functionapp" >> $GITHUB_ENV
+
+    - name: Set environment variables - staging
+      if: github.ref == 'refs/heads/master'
+      shell: bash
+      run: |
+        echo "RESOURCE_GROUP=prime-data-hub-staging" >> $GITHUB_ENV
+        echo "TERRAFORM_ENV=staging" >> $GITHUB_ENV
+        echo "ACR_NAME=pdhstagingcontainerregistry.azurecr.io" >> $GITHUB_ENV
+        echo "IMAGE_REPO=pdhstaging" >> $GITHUB_ENV
+        echo "TAG=${{ github.sha }}" >> $GITHUB_ENV
+        echo "FUNCTION_APP=pdhstaging-functionapp" >> $GITHUB_ENV
+
+    - name: Set environment variables - production
+      if: github.ref == 'refs/heads/production'
+      shell: bash
+      run: |
+        echo "RESOURCE_GROUP=prime-data-hub-prod" >> $GITHUB_ENV
+        echo "TERRAFORM_ENV=prod" >> $GITHUB_ENV
+        echo "ACR_NAME=pdhprodcontainerregistry.azurecr.io" >> $GITHUB_ENV
+        echo "IMAGE_REPO=pdhprod" >> $GITHUB_ENV
+        echo "TAG=${{ github.sha }}" >> $GITHUB_ENV
+        echo "FUNCTION_APP=pdhprod-functionapp" >> $GITHUB_ENV
+
+    - uses: actions/download-artifact@v2
+      with:
+        name: prime-router-build-${{ github.run_id }}
+        path: prime-router
+
+    - name: Unpack router files
+      working-directory: prime-router
+      run: tar -zxvf prime-router-build.tar.gz
+      shell: bash
+
+    - name: ACR login
+      working-directory: prime-router
+      run: az acr login --name ${{ env.ACR_NAME }}
+      shell: bash
+
+    - name: Build Docker image
+      working-directory: prime-router
+      run: docker build . --file Dockerfile --tag ${{ env.ACR_NAME }}/${{ env.IMAGE_REPO }}:latest  --tag ${{ env.ACR_NAME }}/${{ env.IMAGE_REPO }}:${{ env.TAG }}
+      shell: bash
+
+    - name: Push Docker image
+      run: docker push ${{ env.ACR_NAME }}/${{ env.IMAGE_REPO }} --all-tags
+      shell: bash
+
+    - name: Whitelist GitHub action runner in function app slot
+      id: ip-whitelist-create
+      run: |
+        RUNNER_IP="$(dig +short myip.opendns.com @resolver1.opendns.com)"
+        echo "::add-mask::$RUNNER_IP"
+        az functionapp config access-restriction add -g ${{ env.RESOURCE_GROUP }} -n ${{ env.FUNCTION_APP }} --slot candidate --rule-name GitHubActionIPV4 --action Allow --ip-address $RUNNER_IP --priority 750 > /dev/null 2>&1
```

Comment:
Brainstorming idea: what about adding a `--description` to this line (and the removal below) with some quick blurb about it being a temp ip used to connect github to azure? That way if these rules don't get removed for some reason (e.g. poorly timed outage), then spotting them might be easier? [az doc ref](https://docs.microsoft.com/en-us/cli/azure/functionapp/config/access-restriction?view=azure-cli-latest) Maybe if it's logged somewhere, the description might be available as well?